### PR TITLE
Adding eval-adxl355 example for chibios platform support

### PR DESCRIPTION
**Pull request by rbudai98**
* Initial version of integration of ChibiOS with Eval-adxl355 example (suggestions for other integration structure are welcomed)
* For build must have chibios dowloaded from original sources (https://www.chibios.org/dokuwiki/doku.php)
* Variables for make: `PLATFORM=chibios`, `CHIBIOS=<path toward the downloaded files>`, `CHIBIOS_BSP=chibios_cfg/board.mk`
(the last variable is a separation of the general chibios makefile, from the board related files, in order to allow the user to add later different boards to run the example)
!NOTE: in current implementation the PROJECT variable and the CFLAGS variablre has to be changed to PROJECT_CH and CFLAGS_CH, within the os filesystem, in order not to mix the makefile rules

### CODE REVIEW

File: projects/eval-adxl355-pmdz/Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Select the example you want to enable by chossing y for enabling and n for disabling
-DUMMY_EXAMPLE = n
+DUMMY_EXAMPLE = y
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/Makefile:2]: these flags should be modified in the Makefile.

File: projects/eval-adxl355-pmdz/src/platform/stm32/parameters.h [hunk 2]
```diff
@@ -74,8 +74,8 @@ extern UART_HandleTypeDef huart5;
 #define SPI_OPS         &stm32_spi_ops
 #define SPI_EXTRA       &adxl355_spi_extra_ip
 
-extern struct stm32_uart_init_param adxl355_uart_extra_ip;
-extern struct stm32_spi_init_param adxl355_spi_extra_ip;
+extern struct chibios_uart_init_param adxl355_uart_extra_ip;
```
Comment (rbudai98) [inline projects/eval-adxl355-pmdz/src/platform/stm32/parameters.h:77]: not needed, will discharge the changes in the next FP

File: projects/eval-adxl355-pmdz/Makefile (hunk 1 above)
Comment (amiclaus) [reply projects/eval-adxl355-pmdz/Makefile:2]: they shouln't be added in another file. you should call from the command line with these flags. similar to what is done in builds.json

Comment (rbudai98) [reply projects/eval-adxl355-pmdz/Makefile:2]: okay, than will just discard the changes

File: drivers/platform/chibios/chibios_alloc.c [hunk 3]
```diff
@@ -37,9 +37,9 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *******************************************************************************/
 
-#include "no_os_alloc.h"
 #include "hal.h"
 #include <string.h>
+#include "no_os_alloc.h"
```
Comment (amiclaus) [inline drivers/platform/chibios/chibios_alloc.c:42]: unrelated change.

File: projects/eval-adxl355-pmdz/builds.json [hunk 4]
```diff
@@ -1,88 +1,93 @@
 {
+  "chibios": {
+    "dummy_example": {
+      "flags": "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n TARGET=CHIBIOS ADXL355=y"
+    }
+  },
   "stm32": {
     "dummy_example": {
-      "flags" : "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n ADXL355=y"
+      "flags": "DUMMY_EXAMPLE=y IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=n ADXL355=y"
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/builds.json:9]: please fix this file, there are a lot of unrelated changes.

File: projects/eval-adxl355-pmdz/builds.json [hunk 5]
```diff
@@ -1,88 +1,93 @@
...
-      "flags" : "DUMMY_EXAMPLE=n IIO_EXAMPLE=y IIO_TRIGGER_EXAMPLE=n ADXL355=y"
+      "flags": "DUMMY_EXAMPLE=n IIO_EXAMPLE=y IIO_TRIGGER_EXAMPLE=n ADXL355=y"
     },
     "iio_trigger_example": {
-      "flags" : "DUMMY_EXAMPLE=n IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=y ADXL355=y"
+      "flags": "DUMMY_EXAMPLE=n IIO_EXAMPLE=n IIO_TRIGGER_EXAMPLE=y ADXL355=y"
     }
   }
-}
+}
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/builds.json:93]: nit: no new line

File: projects/eval-adxl355-pmdz/chibios_cfg/board_cfg.mk [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+include $(CHIBIOS)/os/common/startup/ARMCMx/compilers/GCC/mk/startup_stm32f4xx.mk
+
+include $(CHIBIOS)/os/hal/ports/STM32/STM32F4xx/platform.mk
+include $(CHIBIOS)/os/hal/boards/ADI_EVAL_SDP_CK1Z/board.mk
+
+include $(CHIBIOS)/os/common/ports/ARMv7-M/compilers/GCC/mk/port.mk
+
+# Define linker script file here
+LDSCRIPT= $(STARTUPLD)/STM32F469xI.ld
+
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/chibios_cfg/board_cfg.mk:14]: nit: extra line.

File: projects/eval-adxl355-pmdz/src.mk [hunk 7]
```diff
@@ -35,9 +35,7 @@ SRCS += $(DRIVERS)/api/no_os_gpio.c \
...
 		$(DRIVERS)/api/no_os_uart.c \
 		$(NO-OS)/util/no_os_list.c \
-		$(NO-OS)/util/no_os_util.c \
-		$(NO-OS)/util/no_os_alloc.c
+		$(NO-OS)/util/no_os_util.c 
 
 INCS += $(DRIVERS)/accel/adxl355/adxl355.h
-SRCS += $(DRIVERS)/accel/adxl355/adxl355.c
-
+SRCS += $(DRIVERS)/accel/adxl355/adxl355.c
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src.mk:41]: unrelated changes

File: projects/eval-adxl355-pmdz/src/platform/aducm3029/platform_src.mk [hunk 8]
```diff
@@ -8,4 +8,6 @@ endif
 SRC_DIRS += $(PLATFORM_DRIVERS)
 SRC_DIRS += $(INCLUDE)
 
+SRCS += $(NO-OS)/util/no_os_alloc.c
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/aducm3029/platform_src.mk:11]: i don't think this should be added here.

File: projects/eval-adxl355-pmdz/src/platform/chibios/main.c [hunk 9]
```diff
@@ -0,0 +1,101 @@
+/*
+    ChibiOS - Copyright (C) 2006..2018 Giovanni Di Sirio
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/chibios/main.c:5]: inappropriate header comment for the main source file.

File: projects/eval-adxl355-pmdz/src/platform/chibios/main.c [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+
+#if (DUMMY_EXAMPLE + IIO_EXAMPLE + IIO_TRIGGER_EXAMPLE == 0)
+#error At least one example has to be selected using y value in Makefile.
+#elif (DUMMY_EXAMPLE + IIO_EXAMPLE + IIO_TRIGGER_EXAMPLE > 1)
+#error Selected example projects cannot be enabled at the same time. \
+Please enable only one example and re-build the project.
+#endif
+
+	return ret;
+}
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/chibios/main.c:101]: nit: new line

File: projects/eval-adxl355-pmdz/chibios_cfg/halconf.h [hunk 11]
```diff
@@ -0,0 +1,550 @@
+/*
+    ChibiOS - Copyright (C) 2006..2020 Giovanni Di Sirio
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/chibios_cfg/halconf.h:5]: is the `chibios_cfg` folder subject for review? or are these files autogenerated?

File: projects/eval-adxl355-pmdz/src/platform/chibios/parameters.c [hunk 12]
```diff
@@ -0,0 +1,75 @@
+/***************************************************************************//**
+ *   @file   parameters.c
+ *   @brief  Definition of chibios platform data used by eval-adxl355-pmdz project.
+ *   @author Robert Budai (robert.budai@analog.com)
+********************************************************************************
+ * Copyright 2022(c) Analog Devices, Inc.
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/chibios/parameters.c:6]: 2023

File: projects/eval-adxl355-pmdz/src/platform/pico/platform_src.mk [hunk 13]
```diff
@@ -12,3 +12,5 @@ SRCS += $(PLATFORM_DRIVERS)/pico_delay.c    \
         $(PLATFORM_DRIVERS)/pico_spi.c      \
         $(PLATFORM_DRIVERS)/pico_timer.c    \
         $(PLATFORM_DRIVERS)/pico_uart.c
+
+SRCS += $(NO-OS)/util/no_os_alloc.c
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/pico/platform_src.mk:16]: new line

File: projects/eval-adxl355-pmdz/src/platform/stm32/platform_src.mk [hunk 14]
```diff
@@ -14,3 +14,5 @@ SRCS += $(PLATFORM_DRIVERS)/stm32_delay.c     \
 	$(PLATFORM_DRIVERS)/stm32_gpio_irq.c  \
 	$(PLATFORM_DRIVERS)/stm32_uart.c      \
 	$(PLATFORM_DRIVERS)/stm32_uart_stdio.c
+
+SRCS += $(NO-OS)/util/no_os_alloc.c
```
Comment (amiclaus) [inline projects/eval-adxl355-pmdz/src/platform/stm32/platform_src.mk:18]: new line

File: tools/scripts/generic.mk [hunk 15]
```diff
@@ -433,3 +437,4 @@ list:
 
 # Adding header files dependancy on object files
 -include $(OBJS:.o=.d)
+endif
```
Comment (amiclaus) [inline tools/scripts/generic.mk:440]: new line

**Review by amiclaus — COMMENTED** [review]
some preliminary comments.

File: projects/eval-adxl355-pmdz/src/platform/aducm3029/platform_src.mk (hunk 8 above)
Comment (rbudai98) [reply projects/eval-adxl355-pmdz/src/platform/aducm3029/platform_src.mk:11]: Moved here because the memory allocation is different compared to the rest of the platforms

File: projects/eval-adxl355-pmdz/chibios_cfg/halconf.h (hunk 11 above)
Comment (rbudai98) [reply projects/eval-adxl355-pmdz/chibios_cfg/halconf.h:5]: no, the config files shouldn't be relevant for review, however structural suggestions are welcomed

**Comment by rbudai98** [comment]
v2:
* applied suggestion on code styling
* generic.mk: moved if prior to any platform related check, in order to exclude extra checks
